include/zephyr/drivers/spi.h

```diff
@@ -152,24 +152,49 @@ extern "C" {
 /**
  * @brief SPI Chip Select control structure
  *
- * This can be used to control a CS line via a GPIO line, instead of
- * using the controller inner CS logic.
+ * This describes how the SPI CS line should be controlled, including whether
+ * to use GPIO or native hardware CS, and timing parameters.
  *
  */
 struct spi_cs_control {
```

Comment:
Perhaps it's time to rename this to spi_cs_and_delay_control

Reply:
renaming would be an API change instead of an extension though, are you sure it's worth doing that, as opposed to just having a couple extra bytes in the overall spi_config struct (whether the way I have in this PR or introduce a new spi_timing struct)

Reply:
If we need an API change (whatever that could be), so be it. Better having a sane API rather than a half-hacked one.